x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_color_property.tsx

```diff
@@ -145,81 +145,59 @@ export class DynamicColorProperty extends DynamicStyleProperty<ColorDynamicOptio
   }
 
   _getOrdinalColorMbExpression() {
-    const targetName = this.getFieldName();
+    let colorStops: Array<number | string> | null = null;
     if (this._options.useCustomColorRamp) {
       if (!this._options.customColorRamp || !this._options.customColorRamp.length) {
         // custom color ramp config is not complete
         return null;
       }
-
-      const colorStops: Array<number | string> = this._options.customColorRamp.reduce(
-        (accumulatedStops: Array<number | string>, nextStop: OrdinalColorStop) => {
-          return [...accumulatedStops, nextStop.stop, nextStop.color];
-        },
-        []
-      );
-      const firstStopValue = colorStops[0] as number;
-      const lessThanFirstStopValue = firstStopValue - 1;
-      return [
-        'step',
-        ['coalesce', [this.getMbLookupFunction(), targetName], lessThanFirstStopValue],
-        RGBA_0000, // MB will assign the base value to any features that is below the first stop value
-        ...colorStops,
-      ];
-    }
-
-    if (this.getDataMappingFunction() === DATA_MAPPING_FUNCTION.PERCENTILES) {
+      colorStops = this._getCustomRampColorStops();
+    } else if (this.getDataMappingFunction() === DATA_MAPPING_FUNCTION.PERCENTILES) {
       const percentilesFieldMeta = this.getPercentilesFieldMeta();
       if (!percentilesFieldMeta || !percentilesFieldMeta.length) {
         return null;
       }
-
-      const colorStops = getPercentilesMbColorRampStops(
+      colorStops = getPercentilesMbColorRampStops(
         this._options.color ? this._options.color : null,
         percentilesFieldMeta
       );
-      if (!colorStops) {
+    } else {
+      const rangeFieldMeta = this.getRangeFieldMeta();
+      if (!rangeFieldMeta) {
         return null;
       }
-
-      const lessThanFirstStopValue = percentilesFieldMeta[0].value - 1;
-      return [
-        'step',
-        ['coalesce', [this.getMbLookupFunction(), targetName], lessThanFirstStopValue],
```

Comment:
With this PR, "makeMbClampedNumberExpression" is now used with percentiles for clamping values below the first stop to the first stop. This has no effect since percentiles uses the 0th percentile so all values are within the provided domain range.